ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -401,167 +455,369 @@ void PowerSystemNode::injectCustomFault(bool& fault_activated,
     else
     {
       auto data = sequence[index];
-      wattage += data[MessageId::Watts];
-      voltage += data[MessageId::Volts];
-      temperature += data[MessageId::Centigrade];
-      index += m_profile_increment;
+
+      // Evenly distribute the power draw, voltage, and temperature values from
+      // the custom fault profile across each model.
+      // NOTE this behavior may need to be updated in the future (e.g. unsure
+      // exactly how voltage and temperature affect predictions, as they do not
+      // directly add to the values like power does).
+      double wattage = data[MessageId::Watts] / NUM_MODELS;
+      double voltage = data[MessageId::Volts] / NUM_MODELS;
+      double temperature = data[MessageId::Centigrade] / NUM_MODELS;
```

Comment:
This is part of the fault injection function, I believe. The expected behavior there isn't really properly defined because the voltage and temperature parts of input data don't seem to directly influence GSAP's predictions. When we send in additional wattage, it is immediately reflected in GSAP's next prediction (RUL decreases proportionally & SoC starts to drain faster), but when we send in additional voltage or temperature, it doesn't seem to even change the prediction temperature by much if at all.

You're probably right that this fault injection isn't correct behavior, but I need to talk with Chetan to confirm what it should look like.